### PR TITLE
[Fix] RenderComponent.material returns assigned material for primitives

### PR DESCRIPTION
**Pull request by mvaligursky**
Fixes https://github.com/playcanvas/engine/issues/3664

RenderComponent has material property, used to access a single material in case of non-asset render. This was up  not being set during material loading.

### CODE REVIEW

File: src/framework/components/render/component.js [hunk 1]
```diff
@@ -682,7 +692,8 @@ class RenderComponent extends Component {
         });
     }
 
-    set materialAssets(value = []) {
+    set materialAssets(value) {
+        value = value || [];
```
Comment (slimbuck) [inline src/framework/components/render/component.js:696]: was this change intentional?

Comment (willeastcott) [reply src/framework/components/render/component.js:696]: Yeah, I thought this looked odd too. If there is some reason why default parameters shouldn't be used in a setter, I would at least recommend a comment because otherwise, somebody will just switch it back. :smile:

Comment (mvaligursky) [reply src/framework/components/render/component.js:696]: to me based on looking online it seems ES6 does not support default parameters in setters.
Our lint flagged it as an issue.

Comment (mvaligursky) [reply src/framework/components/render/component.js]: I tested it and the default value for setter works. When I assign undefined to it, it uses the default value instead. It seems the warning is only for TS, not for JS - can it be globally hidden?
Screenshot 2021-11-05 at 09 12 47

**Review by slimbuck — APPROVED** [review]
